auction-app-frontend/src/pages/AddItem/AddItem.jsx

```diff
@@ -0,0 +1,169 @@
+import { useState, useEffect } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+
+import { ProductDetailsForm, ProductPriceForm, LocationForm } from "src/components";
+
+import { ADD_ITEM_FORMS_MAP, ROUTE_PATHS } from "src/constants";
+import { addProduct } from "src/services";
+import { useUser } from "src/store/UserContext";
+
+import axios from 'axios';
+
+import "./style.scss";
+
+const AddItem = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const { userId } = useUser();
+
+  const initialForm = location.hash ? location.hash.replace("#", "") : ADD_ITEM_FORMS_MAP.DETAILS;
+
+  const [activeForm, setActiveForm] = useState(initialForm);
+  const [formData, setFormData] = useState({
+    details: {},
+    prices: {},
+    shipping: {}
+  });
+
+  const handleHashChange = () => {
+    const newForm = location.hash.replace("#", "");
+    
+    if (newForm && newForm !== activeForm) {
+      setActiveForm(newForm);
+    }
+  };
+
+  useEffect(() => {
+    navigate(`#${activeForm}`);
+  }, []);
+
+  useEffect(() => {
```

Comment:
Can you please explain me logic here?
Why do we do this?

Reply:
i am using a hashchange event listener to respond to hash changes in the url and update the activeForm accordingly